src/ray/core_worker/transport/direct_actor_task_submitter.cc

```diff
@@ -562,6 +567,17 @@ void CoreWorkerDirectActorTaskSubmitter::HandlePushTaskReply(
       RAY_CHECK(is_retryable_exception);
       error_info = gcs::GetRayErrorInfo(rpc::ErrorType::TASK_EXECUTION_EXCEPTION,
                                         reply.task_execution_error());
+    } else if (status.IsStaleTaskError()) {
+      // The task is considered stale by actor. This can heppen when the actor receives
+      // a task, and the connection broke, and the caller resubmits the task with the
+      // same seqno. This task may be retried out of order as if it's retryable user
```

Comment:
I am a little confused about this behavior. Since we reorder the execution based on seqno on the receiver side, it is not possible to retry this out of order right? 

1. If the task has been accepted before connection break, it is ignored, so it is not out-of-order retried (it is simply ignored)
2. if the task has not been accepted before connection break, the task with next seqno shouldn't have been executed yet, so the order is preserved

am I missing something here?

Reply:
For case (1) it's now ignored and makes the process hang. So we need to use a new seqno to retry it out of order.